tests/test_documents.py

```diff
@@ -0,0 +1,381 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# This file is formatted with Python Black
+
+import tests as xdp
+
+import pytest
+import dbus
+from pathlib import Path
+import os
+from gi.repository import GLib
+
+
+def path_from_null_term_bytes(bytes):
+    path_bytes = bytes.split(b"\x00")[0]
+    return Path(os.fsdecode(path_bytes))
```

Comment:
Shouldn't this assert that the path contains exactly one `\0`, and it is the last byte?

Reply:
I'm unsure. In C you'd pass around the pointer and everything up to the first `\0` is what things would work with, and the python code here mimics that behavior.